extension/public/reddit.js

```diff
@@ -5,95 +5,117 @@ var Reddit = {
+  //updates new visible posts to be filtered
+  checkForUpdate: function(){
+      document.addEventListener('scroll', (ev) => {
+      this.counter++;
+      if(this.counter == this.refreshFrequency){
+          this.counter = 0;
+          this.blurImage();
+          console.log("updated");
+      }
+      });
   },
 
-  // initalize all UI elements and onclick binding
-  setupUI: function () {
-    list = document.querySelector(this.listSelector);
-    list.innerHTML = this.dashboardHTML() + list.innerHTML;
 
-    document
-      .querySelector(".reddit-control.unhide_links")
-      .addEventListener("click", () => {
-        this.restoreDOM();
-        console.log("unhide all");
-      });
 
-    document
-      .querySelector(".reddit-control.hide_links")
-      .addEventListener("click", () => {
-        this.setupPostUI();
-        console.log("all links hidden");
-      });
+  //check if any configured keywords are found in a post
+  //return true if found
+  async filterText (post,callBack) {
+      //temporary filter data
+      //these data must be stored in cookie
+      const blockedWordsData = [
+          "denr",
+          "nobody",
+          "taranaki",
+          "covid",
+          "isis",
+          "pizza"
+      ];
+
+      const title = this.getDOMTitle(post);
+      const description = this.getDOMDescription(post);
+      //image varible contain image url then contain result of ocr
+      let image = this.getDOMImageLink(post);
+
+      if(image != "n/a"){
+          image = await this.ocr(image);
+          image = image.toLowerCase();
+          //callback to wait ocr to be processed
+          callBack();
+      }
+
+      console.log(image);
+      //search if blockedwords are in the title
+      //true if found
+      for (i in blockedWordsData){
+          console.log("pass");
+          if(title.search(blockedWordsData[i]) != -1){
+            return true;
+          }else if (image.search(blockedWordsData[i]) != -1){
+            return true;
+          }else if (description.search(blockedWordsData[i]) != -1){
+            return true;
+          }
+      }
+      return false;
   },
 
-  setupPostUI: function () {
-    // adds hide button to each Reddit post
-    posts = document.querySelectorAll(this.linkSelector);
-    console.log(posts);
-    posts.forEach((post) => {
-      post.innerHTML += this.hideLinkHTML();
-      post
-        .querySelector(".reddit-action.hide")
-        .addEventListener("click", (event) => {
-          this.removeLink(event);
-        });
-    });
-    console.log("setup post UI");
-  },
+  //extract text from image using tesseract
+  //returns text
+  async ocr (url) {
+
+      const { createWorker } = Tesseract;
+      const worker = createWorker ({
+      logger: m => console.log(m)
+      });
+
+      await worker.load();
+      await worker.loadLanguage('eng');
+      await worker.initialize('eng');
+      const { data: { text } } = await worker.recognize(url);
```

Comment:
Is it worth creating an issue for setting up multiple language support?

Reply:
What is URL? Is that the actual byte-data of the image, a reference to it in the DOM, or the actual URI e.g. https://....?

If it's URI we should pass the data directly instead of requiring tesseract to perform a second download of the image. If it's a DOM tree this variable needs to be renamed.

Reply:
this is URL, using return data of function getDOMImageLink
Ryan has implemented the getDOMImageLink to return the URL from the DOM.
This could be resolved in a further issue.( speed up OCR )